EIPS/eip-5792.md

```diff
@@ -0,0 +1,257 @@
+      function signature and parameters.
+    - `optional`: `BOOLEAN` - (optional, default false) Whether the wallet should continue with subsequent messages if
+      this particular message fails
+
+##### Example
+
+```json
+[
+  {
+    "chainId": 1,
+    "from": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+    "messages": [
+      {
+        "to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+        "gas": "0x76c0",
+        "value": "0x9184e72a",
+        "data": "0xd46e8dd67c5d32be8d46e8dd67c5d32be8058bb8eb970870f072445675058bb8eb970870f072445675",
+        "optional": true
+      },
+      {
+        "to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
+        "gas": "0xdefa",
+        "value": "0x182183",
+        "data": "0xfbadbaf01"
+      }
+    ]
+  }
+]
+```
+
+#### Return value
+
+DATA, 32 Bytes - an identifier that the wallet uses to represent the bundle of messages. This identifier MUST persist
+between connections of the same Dapp and account.
+
+The dapp may track this identifier to check the status of the transaction in the future.
+The returned identifier value MUST be unique for the given `from` address.
+The length of the returned response MUST NOT exceed 66 characters.
+There are no other requirements on the format of the identifier.
+
+##### Example
+
+The identifier may be the hash of the message bundle, e.g.:
+
+```json
+"0xe670ec64341771606e55d6b4ca35a1a6b75ee3d5145a99d05921026d1527331"
+```
+
+The identifier may be a numeric identifier represented as a string, e.g.:
+
+```json
+"0x01"
+```
+
+### `wallet_getBundleStatus`
+
+Returns the status of a bundle that was sent via `wallet_sendMessageBundle`. The identifier of the bundle is the value
+returned from the `wallet_sendMessageBundle` RPC. Note this method only returns a subset of fields
+that `eth_getTransactionReceipt` returns, excluding any fields that may differ in meaning between wallet
+implementations.
+
+#### Parameters
+
+The method takes an array containing one string element, a bundle ID returned by `wallet_sendMessageBundle`.
+
+##### Example
+
+```json
+[
+  "0xe670ec64341771606e55d6b4ca35a1a6b75ee3d5145a99d05921026d1527331"
+]
+```
+
+#### Return value
+
+Returns an object describing the status of each message in the bundle. Each message has its own status in the response,
+even if all the messages were submitted as part of a single transaction.
+
+- `id`: `DATA` - The identifier of the transaction returned by `wallet_sendMessageBundle`
+- `messages`: `ARRAY` - An array of objects containing the status of each message from the bundle
+    - `status`: `"PENDING" | "CONFIRMED"` - Either `"PENDING"` indicating the message is not yet included on-chain,
+      or `"CONFIRMED"` indicating the message was included in a block
+    - `receipt`: `OBJECT` - (optional) An object containing the receipt for the given message, _iff_ status
+      is `"CONFIRMED"`
+        - `logs`: `ARRAY` - An array of log objects describing logs that were emitted by the message
```

Comment:
I think it will be non-trivial to determine this in the case of a batched tx in a wallet since many logs may be emitted during execution and you'd need a trace to determine which subcall it belongs to?

Reply:
It's not difficult, but the implementation differs based on the wallet. You don't have to trace if you just emit a separator log between each call.